tools/build-templated-pages/src/features.rs

```diff
@@ -0,0 +1,110 @@
+use std::{cmp::Ordering, fs::File};
+
+use serde::Serialize;
+use tera::{Context, Tera};
+use toml_edit::Document;
+
+use crate::Command;
+
+#[derive(Debug, Serialize, PartialEq, Eq)]
+struct Feature {
+    name: String,
+    description: String,
+    is_default: bool,
+}
+
+impl Ord for Feature {
+    fn cmp(&self, other: &Self) -> Ordering {
+        self.name.cmp(&other.name)
+    }
+}
+
+impl PartialOrd for Feature {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+fn parse_features(panic_on_missing: bool) -> Vec<Feature> {
+    let manifest_file = std::fs::read_to_string("Cargo.toml").unwrap();
+    let manifest = manifest_file.parse::<Document>().unwrap();
+
+    let features = manifest["features"].as_table().unwrap();
+    let default: Vec<_> = features
+        .get("default")
+        .unwrap()
+        .as_array()
+        .unwrap()
+        .iter()
+        .flat_map(|v| {
+            std::iter::once(v.as_str().unwrap().to_string()).chain(
+                features
+                    .get(v.as_str().unwrap())
+                    .unwrap()
+                    .as_array()
+                    .unwrap()
+                    .iter()
+                    .map(|v| v.as_str().unwrap().to_string()),
+            )
+        })
+        .collect();
```

Comment:
This can be simplified a bit by defining a closure like so:

```rust
let array_from_table = |table: &Table, key: &str| table.get(key).unwrap().as_array().unwrap();
```

This way you got:

```rust
array_from_table(features, "default").iter().flat_map(|v| {
  std::iter::once(v.as_str().unwrap().to_string()).chain(
    array_from_table(features, v.as_str().unwrap())
      .iter()
      .map(|v| v.as_str().unwrap().to_string()),
  )
})
```
Also I believe it's possible to `default: Vec<&toml::value::Value>`, since `Value` implements `PartialEq<Value>` it can be compared still.

```rust
array_from_table(features, "default").iter().flat_map(|v| {
  std::iter::once(v).chain(
    array_from_table(features, v.as_str().unwrap()).iter()
  )
})
```

Reply:
I couldn't get those to work

Reply:
That's a shame. I'll approve the PR. The code is alright (since I could understand it) but invariably someone in 6 months will see this and get angry at it.